app/helpers/injection_helper.rb

```diff
@@ -79,73 +79,75 @@ def inject_current_order_cycle
                     Api::PaymentMethodSerializer, current_order: current_order
   end
 
   def inject_taxons
-    inject_json_ams "taxons", Spree::Taxon.all.to_a, Api::TaxonSerializer
+    inject_json_list "taxons", Spree::Taxon.all.to_a, Api::TaxonSerializer
   end
 
   def inject_properties
-    inject_json_ams "properties", Spree::Property.all.to_a, Api::PropertySerializer
+    inject_json_list "properties", Spree::Property.all.to_a, Api::PropertySerializer
   end
 
   def inject_currency_config
-    inject_json_ams "currencyConfig", {}, Api::CurrencyConfigSerializer
+    inject_json "currencyConfig", {}, Api::CurrencyConfigSerializer
   end
 
   def inject_open_street_map_config
-    inject_json_ams "openStreetMapConfig", {}, Api::OpenStreetMapConfigSerializer
+    inject_json "openStreetMapConfig", {}, Api::OpenStreetMapConfigSerializer
   end
 
   def inject_spree_api_key
     render partial: "json/injection_ams", locals: { name: 'spreeApiKey', json: "'#{@spree_api_key}'" }
   end
 
   def inject_available_countries
-    inject_json_ams "availableCountries", available_countries, Api::CountrySerializer
+    inject_json_list "availableCountries", available_countries, Api::CountrySerializer
   end
 
   def inject_enterprise_attributes
     render partial: "json/injection_ams", locals: { name: 'enterpriseAttributes', json: @enterprise_attributes.to_json.to_s }
   end
 
   def inject_orders
-    inject_json_ams "orders", @orders.all, Api::OrderSerializer
+    inject_json_list "orders", @orders.all, Api::OrderSerializer
   end
 
   def inject_shops
     customers = spree_current_user.customers
     shops = Enterprise.where(id: @orders.pluck(:distributor_id).uniq | customers.pluck(:enterprise_id))
-    inject_json_ams "shops", shops.all, Api::ShopForOrdersSerializer
+    inject_json_list "shops", shops.all, Api::ShopForOrdersSerializer
   end
 
   def inject_saved_credit_cards
     data = spree_current_user ? spree_current_user.credit_cards.with_payment_profile.all : []
 
-    inject_json_ams "savedCreditCards", data, Api::CreditCardSerializer
+    inject_json_list "savedCreditCards", data, Api::CreditCardSerializer
   end
 
   def inject_current_user
-    inject_json_ams "user", spree_current_user, Api::UserSerializer
+    inject_json "user", spree_current_user, Api::UserSerializer
   end
 
   def inject_rails_flash
-    inject_json_ams "railsFlash", OpenStruct.new(flash.to_hash), Api::RailsFlashSerializer
+    inject_json "railsFlash", OpenStruct.new(flash.to_hash), Api::RailsFlashSerializer
   end
 
-  def inject_json_ams(name, data, serializer, opts = {})
-    if data.is_a?(Array)
-      opts = { each_serializer: serializer }.merge(opts)
-      serializer = ActiveModel::ArraySerializer
-    end
+  def inject_json_list(name, data, serializer, opts = {})
+    opts = { each_serializer: serializer }.merge(opts)
+    serializer = ActiveModel::ArraySerializer
 
+    inject_json(name, data, serializer, opts)
```

Comment:
here.
I havent checked the build yet, but I think we can just do data.to_a here.